modules/core/shared/src/main/scala/io/circe/JsonNumber.scala

```diff
@@ -264,20 +273,20 @@ final object JsonNumber {
     if (result.eq(null)) None else Some(JsonBiggerDecimal(result))
   }
 
+  private[this] val bigDecimalMinLong: JavaBigDecimal = new JavaBigDecimal(Long.MinValue)
+  private[this] val bigDecimalMaxLong: JavaBigDecimal = new JavaBigDecimal(Long.MaxValue)
+
+  private[circe] def bigDecimalIsWhole(value: JavaBigDecimal): Boolean =
+    value.signum == 0 || value.scale <= 0 || value.stripTrailingZeros.scale <= 0
```

Comment:
Might want to also check `value.compareTo(JavaBigDecimal.ZERO) == 0` due to http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6480539